cobra/util/solver.py

```diff
@@ -31,7 +33,10 @@
            for match in dir(optlang) if "_interface" in match}
 
 # Defines all the QP solvers implemented in optlang.
-qp_solvers = ["cplex"]  # QP in gurobi not implemented yet
+qp_solvers = ["cplex", "gurobi"]
+
+# optlang solution statuses which still allow retrieving primal values
+has_primals = [NUMERIC, FEASIBLE, INFEASIBLE, SUBOPTIMAL, ITERATION_LIMIT]
```

Comment:
I wonder if this is something that should be tackled in optlang. `has_primals` (and `has_duals`?) could be an attribute on the `Model` class.

Reply:
Yeah that would be much cleaner...

Reply:
I'll open an issue at optlang but will merge this for now in order to keep moving.